receiver/mongodbatlasreceiver/factory.go

```diff
@@ -82,5 +122,21 @@ func createDefaultConfig() config.Receiver {
 		Alerts: AlertConfig{
 			Enabled: defaultAlertsEnabled,
 		},
+		Logs: LogConfig{
+			Enabled:  true,
+			Projects: []*Project{},
+		},
+	}
+}
+
+// getReceiver ensures we only create a single receiver because it is shared between logs and metrics
+func getReceiver(params component.ReceiverCreateSettings, cfg *Config) (*receiver, error) {
+	var err error
+
+	// If we don't have one yet create it
+	if mongoDBReciever == nil {
+		mongoDBReciever, err = newMongoDBAtlasReciever(params, cfg)
 	}
+
+	return mongoDBReciever, err
 }
```

Comment:
This will cause problems if I make 2 mongoDB atlas receivers with different configurations

e.g:
```yaml
receivers:
  mongodbatlas/1:
    ...
  mongodbatlas/2:
    ...
```
receiver 1 & 2 should both be able to run with different configurations, but this pattern does not allow that.

Is this singleton pattern necessary currently? I know the collector removed this constraint a few versions back, does it cause any issues to let a new instance be created each time?

If we do need this, we need to create a map of receiver ID -> receiver and return the receiver that corresponds to the ID if it was already created.